From da3359a66745a354a43c654d42cfa85430ca70cf Mon Sep 17 00:00:00 2001
From: Tian Yang <tiany@codeaurora.org>
Date: Tue, 27 Apr 2021 13:33:56 -0700
Subject: [PATCH] ip6_tunnel: Fix NLA_F_NESTED check issue in kernel 5.4

Use nla_parse_nested_deprecated instead of nla_parse_nested to bypass the NLA_F_NESTED flag check.

Change-Id: Ic2bc0903bd5366f516be011d87932138ede1d004
Signed-off-by: Tian Yang <tiany@codeaurora.org>
---
 net/ipv6/ip6_tunnel.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/net/ipv6/ip6_tunnel.c
+++ b/net/ipv6/ip6_tunnel.c
@@ -2229,7 +2229,7 @@ static void ip6_tnl_netlink_parms(struct
 			struct nlattr *fmrd[IFLA_IPTUN_FMR_MAX + 1], *c;
 			struct __ip6_tnl_fmr *nfmr;
 
-			nla_parse_nested(fmrd, IFLA_IPTUN_FMR_MAX,
+			nla_parse_nested_deprecated(fmrd, IFLA_IPTUN_FMR_MAX,
 				fmr, ip6_tnl_fmr_policy, NULL);
 
 			if (!(nfmr = kzalloc(sizeof(*nfmr), GFP_KERNEL)))
